test/unit/helpers/form_helper_test.rb

```diff
@@ -0,0 +1,78 @@
+    should "return an unordered list with errors when there are errors present" do
+      expected = "<ul class=\"help-block error-block\"><li>One</li><li>Two</li><li>Three</li></ul>"
+
+      assert_equal expected, form_errors(%w[One Two Three])
+    end
+
+    should "return an empty unordered list when there are no errors present" do
+      expected = "<ul class=\"help-block error-block\"></ul>"
+
+      assert_equal expected, form_errors
+    end
+  end
+
+  context "form_group" do
+    setup do
+      object = stub(field_name: "", errors: Hash.new([]))
+      @form = FormBuilder.new("edition", object, self, {})
+      @label = '<div class="form-label"><label for="edition_field_name">Field name</label></div>'
+      @wrapped_field = '<div class="form-wrapper"><input type="text" value="" name="edition[field_name]" id="edition_field_name" /></div>'
+      @error_block = '<ul class="help-block error-block"></ul>'
+    end
+
+    should "return a form-group html structure without: errors, help block, custom label or custom attributes" do
+      output = form_group(@form, :field_name) { @form.text_field(:field_name) }
+
+      assert_equal %(<div class="form-group">#{@label}#{@wrapped_field}#{@error_block}</div>), output
+    end
+
+    should "return a form-group html structure with a help block when help text is supplied" do
+      output = form_group(@form, :field_name, help: "Help block text") { @form.text_field(:field_name) }
+      help_block = '<div class="help-block">Help block text</div>'
+
+      assert_equal %(<div class="form-group">#{@label}#{@wrapped_field}#{@error_block}#{help_block}</div>), output
+    end
+
+    should "return a form-group html structure with a custom label" do
+      output = form_group(@form, :field_name, label: "Custom label text") { @form.text_field(:field_name) }
+      custom_label = '<div class="form-label"><label for="edition_field_name">Custom label text</label></div>'
+
+      assert_equal %(<div class="form-group">#{custom_label}#{@wrapped_field}#{@error_block}</div>), output
+    end
+
+    should "return a form-group html structure with a custom attributes" do
+      output = form_group(@form, :field_name, attributes: { id: :field_id, class: %i[one two] }) { @form.text_field(:field_name) }
+
+      assert_equal %(<div id="field_id" class="one two form-group">#{@label}#{@wrapped_field}#{@error_block}</div>), output
+    end
+
+    should "return a form-group html structure with a has-error class when any errors are present" do
+      object = stub(field_name: "field_name", errors: Hash.new(["cannot be blank", "must be unique"]))
+      form = FormBuilder.new("edition", object, self, {})
+      label = '<div class="form-label"><div class="field_with_errors"><label for="edition_field_name">Field name</label></div></div>'
+      wrapped_field = '<div class="form-wrapper"><div class="field_with_errors"><input type="text" value="field_name" name="edition[field_name]" id="edition_field_name" /></div></div>'
```

Comment:
We shouldn't be testing these `field_with_errors` classes as their implicit Rails ones that are applied outside our unit of code. I think the only thing we need to test error wise is that form-group gets the extra class. We could also check the errors are displayed, but I don't think we need to do it at a granular HTML level. We can get away with `assert_match "cannot be blank", output`